### PR TITLE
[NVFuser] assert that vectors are the same size in translateSingleWelford

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77010

Before, sometimes out_root.size() < in_root.size(), which would result
in a segfault while accessing out_root[i]. If, instead, we just error
out here, an exception will be thrown and then we'll run the fallback
instead of completely erroring out.

### CODE REVIEW

**Comment by davidberard98** [comment]
cc @jjsjann123 - in case we think the fix to the segfault in https://github.com/pytorch/pytorch/issues/76883 will take some time

**Comment by jjsjann123** [comment]
Ha good catch~ I'll try this with my repro~

But thanks for the assert. Regarding #76883, I'm gonna simply disable squeeze/unsqueeze. Let me push a PR for that~

File: torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp [hunk 1]
```diff
@@ -1930,6 +1930,7 @@ void TranslateApplicableWelford::translateSingleWelford(WelfordOp* welford) {
   //  largely taken from batchnorm cpp benchmark
   auto& in_root = in_val->getRootDomain();
   auto& out_root = out_avg->getRootDomain();
+  TORCH_INTERNAL_ASSERT(in_root.size() == out_root.size());
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp:1933]: Naive comment, looking at the logic below, the check here should be `in_root.size() <= out_root.size()`?
```
for (const auto i : c10::irange(in_root.size())
   ... out_root[i] ...
```

Comment (davidberard98) [reply torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp:1933]: sure, good point - I'll change that.

@jjsjann123 is this PR still useful now that the actual fixes are landed?

**Review by eellison — APPROVED** [review]

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp (hunk 1 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp]: Yes, let's merge this one.

**Pull request closed without merging**